ucsb-cs156-f24/proj-happycows-f24-11
# ALL CRUD operations for courses (admin only)

## Overview

In this PR, I created all CRUD operations for the backend of Courses.
- [x] Create personal dokku dev
- [x] Create database table for Courses (initial DB table for courses has been created. Database files for happycows, including migration file, entity, and repository file have been been added)

- [x] Create CoursesController.java, add Get and POST
- [x] Add GET endpoint for single record in Courses table
- [x] Add PUT endpoint for single record in Courses table
- [x] Add DELETE endpoint for specific record in Courses table
## Screenshots (Optional)
Database table on Dokku:
![Screenshot 2024-11-23 at 3 53 42 PM](https://github.com/user-attachments/assets/de9127c1-5f4f-4980-942c-6cfbbb50b781)

CRUD Endpoints on Swagger:
<img width="968" alt="Screenshot 2024-11-24 at 1 47 25 AM" src="https://github.com/user-attachments/assets/9e5257e7-9fe6-4792-be5f-21bf594e702b">

PUT Endpoint:
<img width="958" alt="Screenshot 2024-11-24 at 1 56 51 AM" src="https://github.com/user-attachments/assets/0951780e-760c-400e-ad43-04404419442c">


POST Endpoint:
<img width="961" alt="Screenshot 2024-11-24 at 1 57 14 AM" src="https://github.com/user-attachments/assets/e1522132-d1cb-404a-a665-10c0b56635e1">

GET SINGLE Endpoint:
<img width="953" alt="Screenshot 2024-11-24 at 1 57 48 AM" src="https://github.com/user-attachments/assets/2ced9a53-ca0e-4baf-9e5f-0e26ba06e991">

GET ALL Endpoint:
<img width="945" alt="Screenshot 2024-11-24 at 2 00 56 AM" src="https://github.com/user-attachments/assets/ecb208b1-2050-41d3-a2f0-bfe344bec20f">

DELETE Endpoint:
<img width="990" alt="Screenshot 2024-11-24 at 1 59 03 AM" src="https://github.com/user-attachments/assets/c351dad1-de4a-4f0b-8f9a-ae44c41e3392">
<img width="967" alt="Screenshot 2024-11-24 at 1 59 19 AM" src="https://github.com/user-attachments/assets/3a5de7c8-1f74-491a-ad71-b144b988bd61">


Pit Test Coverage:

![Screenshot 2024-11-25 at 1 26 47 AM](https://github.com/user-attachments/assets/9740ba14-a186-4016-a40a-7a7d7a3b27f6)

Jacoco tests:
<img width="1136" alt="Screenshot 2024-11-24 at 1 39 41 AM" src="https://github.com/user-attachments/assets/6aa444af-8650-4bf5-8d25-372783baebb7">



## Feedback Request (Optional)

Please give suggestions on the following:
- I have defined 3 variables for the Courses.entity file: id, name, and term. I removed GitHub id, start and endtime, and school. Because the application was geared towards UCSB CHEM students, so there was no need for GitHub id or school variable. Additionally, the commons had a variable for start and endtime, so there was no need to create those again.


## Validation (Optional)

1. Go to this dokku link:  https://happycows-adenjo.dokku-11.cs.ucsb.edu
2. Check that all CRUD Operations work (for Admins and regular users)

## Tests



## Linked Issues

Closes #25
